riscv-asm.md

```diff
@@ -525,6 +527,38 @@ Integers in the subinterval [-2048, -1] can also be passed by their (unsigned) a
 in the interval [0xfffff800, 0xffffffff] on RV32I, and
 in [0xfffffffffffff800, 0xffffffffffffffff] on both RV32I and RV64I.
 
+<a name=fp-literal></a>Floating-point literals
+-------------------
+
+The Assembler supports the same floating-point literal formats as those defined
+in the C and C++ standards (i.e., decimal floating-point literals with decimal
+exponents as well as hexadecimal floating-point literals with binary exponents).
+
+Here are some examples:
+* 3.14159
+* 0.271828e1
+* 0x0.3p-4
+
+NOTE: The detailed format of the floating point immediate value can be
+referenced on [this page](https://en.cppreference.com/w/cpp/language/floating_literal)
+
+Load Floating-point Immediate
+-------------------
+
+RISC-V does not offer a generic pseudoinstruction to load an arbitrary floating
+point immediate value. Instead, a programmer can use the `.float`/`.double`
+directive to declare a floating point immediate value in the source code, and
+then load it into a floating point register using the load global
+pseudoinstruction (`fl{h|s|d|q}`).
```

Comment:
Same comment as @asb : you only mention the `.float` and `.double` directives; what should I use in the FP16 and FP128 cases? Note that `.quad` is already used by GAS. Maybe `.float{16,32,64,128}` would be a more scalable approach?

Reply:
Oh, the reason I document `.float` and `.double` here is because `.float16` is not supported by clang, and `.float128` are not supported for both clang and binutils...

`.quad` is supported by both, I think I should add that to doc.

Reply:
My thought is we should add `.float{16,32,64,128}`  and `.bfloat16` in future, but anyway it out of scope of this PR, let me finish fli stuff first.